students/aminavi/session04/mailroom 2 R1.py

```diff
@@ -6,35 +6,40 @@
 quit_action = "quit"
 
 donor_list = {
-    'William': [653772.32 , 953772.32,100000],
-    'Jeff': [8888777.22 ,877.33],
-    'Paul': [663.23, 43.87 ,1.32],
-    'Mark': [1663.23, 4300.87, 10432.00],
-    'Nigel': [5000000 , 10]
+    'william': [653772.32 , 953772.32,100000],
+    'jeff': [8888777.22 ,877.33],
+    'paul': [663.23, 43.87 ,1.32],
+    'mark': [1663.23, 4300.87, 10432.00],
+    'nigel': [5000000 , 10]
     }
+def sort_key(donor_list):
+    return sum(donor_list.values())
+
+def send_thank_you():
+    name = ''
+    while not name or name.lower() == 'list':
+        name = input("Please enter the full name: ")
+        if name.lower() == 'list':
+            donor_list.setdefault(name.lower(),[])
+    donation = int(input("Please enter the donation amount: "))
+    for i in range(len(donor_list.keys())):
+        if name.lower() in  donor_list.keys():
```

Comment:
Minor: No need to say `donor_list.keys()` in lines 25-26. Just `donor_list` (dict name) will be enough (that's how Python dicts work, with `in` and `len(...)`).